WIP: Reorganize Science section. (#179)
Cantera/cantera-website: pages/science/thermodynamics.rst

.. slug: thermodynamics
.. has_math: true
.. title: Calculating phase and species thermodynamics

.. jumbotron::

   .. raw:: html

      <h1 class="display-3">Calculating thermodynamic properties in Cantera</h1>

   .. class:: lead

      Here, we describe how Cantera uses species and phase information to calculate thermodynamic properties. Thermodynamic properties typically depend on information at both the species and phase levels.

>       - The user must specify a thermodynamic model for each species and provide inputs that inform how species-specific properties are calculated (e.g. as a function of temperature).

Author: > I prefer to avoid Latin abbreviations

👍🏼

> The other thing on this line is it might be nice to clarify which are species-specific properties, since you've clarified which ones are phase-average properties.

I can add some examples (reference Cp, h, and s values as a function of T should cover it).

Regarding giving a sense of how species and phase properties relate: I feel like there's a catch-22, here: anything that I can describe as being _generally_ true here would not be something that a phase model provides (if it is always true, then it can just be implemented generally for all `Phase` models).

>       - The user also selects a phase model. This model describes how the species interact with one another to determine overall phase properties. This includes general :math:`P-v-T` behavior, as well as how species-specific properties are used to calculate phase-average properties such as internal energy, entropy, etc.  

Author: My intent is really as simple as "for example, calculate the phase pressure for a given composition, temperature, and total molar volume." 

I can probably add it as a parenthetical.

.. slug: thermodynamics
.. has_math: true
.. title: Calculating phase and species thermodynamics

.. jumbotron::

   .. raw:: html

      <h1 class="display-3">Calculating thermodynamic properties in Cantera</h1>

   .. class:: lead

      Here, we describe how Cantera uses species and phase information to calculate thermodynamic properties. 
      
   Thermodynamic properties typically depend on information at both the species and phase levels. The user must specify thermodynamic models for both levels, and these selections must be compatible with one another. For instance: one cannot pair certain non-ideal species thermodyamic models with an ideal phase model.

   - The user must specify a thermodynamic model for each species and provide inputs that inform how species properties are calculated. For example, the user specifies how the reference enthalpy and entropy values for each species are calcualted, as a function of temperature.

>    - The user also selects a phase model. This model describes how the species interact with one another to determine phase properties and species specific properties, for a given thermodynamic state. This includes general :math:`P-\hat{v}-T` behavior (for example, calculate the phase pressure for a given molar volume, temperature, and chemical composition), as well as how species-specific properties, such as internal energy, entropy, and others depend on the state variables

Reviewer: ```suggestion
   - The user also selects a phase model. This model describes how the species interact with one another to determine phase properties and species specific properties, for a given thermodynamic state. This includes general :math:`P`-:math:`\hat{v}`-:math:`T` behavior (for example, calculate the phase pressure for a given molar volume, temperature, and chemical composition), as well as how species-specific properties, such as internal energy, entropy, and others depend on the state variables
```

Example: The Ideal Gas Model
============================
For a simple example: in the Ideal Gas model, one might use 7-parameter NASA polynomials to specify the species reference thermodynamic quantities.  These would be used to calculate the reference molar enthalpy :math:`\hat{h}_k^\circ(T)` and entropy :math:`\hat{s}_k^\circ(T)` for a given species :math:`k` as a function of temperature :math:`T`. See the `NASA Polynomials Species Thermo entry </science/science-species.html#the-nasa-7-coefficient-polynomial-parameterization>`__ for more information.

> At the phase level, the Ideal Gas Law provides the P-v-T relationship, called an equation of state. This is used, for example, to calculate the pressure as a function of molar volume :math:`\hat{v}` and temperature :math:`T`:

Reviewer: ```suggestion
At the phase level, the Ideal Gas Law provides the :math:`P`-:math:`\hat{v}`-:math:`T` relationship. The ideal gas law is an example of an equation of state. This is used, for example, to calculate the pressure as a function of molar volume :math:`\hat{v}`, and temperature, :math:`T`:
```

.. math::
   p = \frac{\overline{R}T}{\hat{v}}

> where :math:`\overline{R}` is the Universal Gas Constant. Maxwell's relations are used to derive other thermodynamic properties from the equation of state. With the Ideal Gas phase model, these reduce to rather simple forms. For example, for a species :math:`k`, the Ideal Gas molar internal energy :math:`\hat{u}_k` and entropy :math:`\hat{s}_k` are:

Reviewer: ```suggestion
where :math:`\overline{R}` is the Universal Gas Constant. The `Maxwell relations <https://en.wikipedia.org/wiki/Maxwell_relations>`__ are used to derive other thermodynamic properties from the equation of state. With the Ideal Gas phase model, these reduce to rather simple forms. For example, for a species :math:`k`, the Ideal Gas molar internal energy :math:`\hat{u}_k` and entropy :math:`\hat{s}_k` are:
```

.. math::
   \hat{u}_k = \hat{h}^\circ_k(T) - p\hat{v}

   \hat{s}_k = \hat{s}^\circ_k(T) - \overline{R}\ln\left(\frac{pX_k}{p^\circ}\right)

where :math:`X_k` is the mole fraction of species :math:`k`, and where :math:`p^\circ` is the reference pressure at which the properties :math:`\hat{h}_k^\circ(T)` and :math:`\hat{s}_k^\circ(T)` are known.

Please click either of the cards below for details on the species and phase models available in Cantera:

.. container:: container
   :tagname: section

   .. container:: card-deck

      .. container:: card

         .. container::
            :tagname: a
            :attributes: href=species-thermo.html
                         title=Species

            .. container:: card-header section-card
               :tagname: div

               Species

         .. container:: card-body

            .. container:: card-text

>                The models and equations that Cantera uses to calculate species thermodynamic properties.

Reviewer: ```suggestion
               The models and equations that Cantera uses to calculate species thermodynamic properties, such as the NASA 7-parameter polynomial form.
```

      .. container:: card

         .. container::
            :tagname: a
            :attributes: href=phase-thermo.html
                         title=Phases

            .. container:: card-header section-card
               :tagname: div

               Phases

         .. container:: card-body

            .. container:: card-text

>                The theory behind some of Cantera's phase models.

Reviewer: ```suggestion
               The theory behind some of Cantera's phase models, such as the Ideal Gas Law.
```

      
   Thermodynamic properties typically depend on information at both the species and phase levels. The user must specify thermodynamic models for both levels, and these selections must be compatible with one another. For instance: one cannot pair a non-ideal species thermodyamic model with an ideal phase model.

   - The user must specify a thermodynamic model for each species and provide inputs that inform how species-specific properties are calculated. For example, the user specifies how the reference enthalpy and entropy values for each species are calcualted, as a function of temperature.
   - The user also selects a phase model. This model describes how the species interact with one another to determine phase properties and species specific properties, for a given thermodynamic state. This includes general :math:`P-v-T` behavior (for example, calculate the phase pressure for a given molar volume, temperature, and chemical composition), as well as how species-specific properties, such as internal energy, entropy, and others depend on the state variables

Example: The Ideal Gas Model
============================

> For a simple example: in the Ideal Gas model, one might use 7-parameter NASA polynomials to specify the species thermodynamics.  These would be used to calculate the reference molar enthalpy :math:`h_k^\circ(T)` and entropy :math:`s_k^\circ(T)` for a given species :math:`k` as a function of temperature :math:`T`:

Reviewer: > As far as Cantera is concerned, I don't think there is such a thing as an "ideal gas species model", though

Hmmm, fair enough. Maybe we should choose a different example here, which may work if we limit the equations? Perhaps R-K or P-R?

> For a simple example: in the Ideal Gas model, one might use 7-parameter NASA polynomials to specify the species thermodynamics.  These would be used to calculate the reference molar enthalpy :math:`h_k^\circ(T)` and entropy :math:`s_k^\circ(T)` for a given species :math:`k` as a function of temperature :math:`T`:

Author: @bryanwweber -- re: using R-K or P-R.  

a. Those would also rely on NASA polynomials, most frequently.

b. My intent here was for simplicity.  Show that the `species` model is used to find the "reference state" properties, and that the `phase` model uses this reference properties to then find the actual properties. I feel like Ideal Gas is both the most familiar and computationally "cleanest" way to demonstrate this.

However, it seems like you might be after something else - can you clarify?

> For a simple example: in the Ideal Gas model, one might use 7-parameter NASA polynomials to specify the species thermodynamics.  These would be used to calculate the reference molar enthalpy :math:`h_k^\circ(T)` and entropy :math:`s_k^\circ(T)` for a given species :math:`k` as a function of temperature :math:`T`:

Author: I'm fine going with `\hat`.  Not sure if I've got the chops to configure Mathjax, but I'll give it a look.

> For a simple example: in the Ideal Gas model, one might use 7-parameter NASA polynomials to specify the species thermodynamics.  These would be used to calculate the reference molar enthalpy :math:`h_k^\circ(T)` and entropy :math:`s_k^\circ(T)` for a given species :math:`k` as a function of temperature :math:`T`:

Reviewer: > a. Those would also rely on NASA polynomials, most frequently.

Dang.

> b. My intent here was for simplicity. Show that the species model is used to find the \"reference state\" properties, and that the phase model uses this reference properties to then find the actual properties.

That's actually a really good statement of what I think I'm after!

> I feel like Ideal Gas is both the most familiar and computationally \"cleanest\" way to demonstrate this.

Agreed. 

> However, it seems like you might be after something else - can you clarify?

I'm not totally sure I know, except that I'll know it when I see it 🤪 more seriously, I think this is very close to what I'm thinking of. I think it's tilted a little too far into details in this iteration. My comment was an inaccurate attempt to simplify, and also to connect to the prior point that phase and species thermo models have to be compatible. That last bit, the connection, is probably just asking too much of the text here though, so I think we can drop that attempt. But I do think it needs to be simplified in the direction my comment was trending.

I'll take a look at the configuration for Mathjax when I'm back at my computer in the morning my time 😊 it should be similar to defining a macro in LaTeX, mostly a matter of getting it into the right spot.

> For a simple example: in the Ideal Gas model, one might use 7-parameter NASA polynomials to specify the species thermodynamics.  These would be used to calculate the reference molar enthalpy :math:`h_k^\circ(T)` and entropy :math:`s_k^\circ(T)` for a given species :math:`k` as a function of temperature :math:`T`:

Author: Hmm, so does that mean that the molar volume is `\hat{v}`?  That feels... odd, to me, but it technically the consistent approach.
